master/src/main/scala/org/apache/celeborn/service/deploy/master/Master.scala

```diff
@@ -850,6 +852,22 @@ private[celeborn] class Master(
       logInfo(s"Offered extra $offerSlotsExtraSize slots for $shuffleKey")
     }
 
+    if (authEnabled) {
+      // Pass application registration information to the workers
+      val pbApplicationMeta = PbApplicationMeta.newBuilder()
+        .setAppId(requestSlots.applicationId)
+        .setSecret(secretRegistry.getSecretKey(requestSlots.applicationId))
+        .build()
+      val transportMessage =
+        new TransportMessage(MessageType.APPLICATION_META, pbApplicationMeta.toByteArray)
+
+      slots.keySet().asScala.foreach { worker =>
+        logDebug(s"Sending app registration info to ${worker.host}:${worker.internalPort}")
+        internalRpcEnvInUse.setupEndpointRef(
+          RpcAddress.apply(worker.host, worker.internalPort),
+          RpcNameConstants.WORKER_INTERNAL_EP).send(transportMessage)
```

Comment:
`RequestSlots` is currently sent for each shuffleId, so it can happen multiple times for the same application, which will lead to exception in `SecretRegistryImpl#register`：
```
  public void register(String appId, String secret) {
    // TODO: Persist the secret in ratis. See https://issues.apache.org/jira/browse/CELEBORN-1234
    secrets.compute(
        appId,
        (id, oldVal) -> {
          if (oldVal != null) {
            throw new IllegalArgumentException("AppId " + appId + " is already registered.");
          }
          return secret;
        });
  }
```
Further more, in the future the same shuffleId might also send multiple `RequestSlots`. Maybe we can just check whether secret is the same instead of throwing Exception, or avoid sending duplicate `APPLICATION_META` in Master.

Reply:
In order to avoid sending duplicate `APPLICATION_META` from Master to Workers, the master will have to maintain a state that contains which worker has been informed per application. 
Instead, I can fix this on the Worker side, by ignoring the `APPLICATION_META` if the application is already registered.

Reply:
Made the change. PTAL.

Reply:
Thanks for this change. Referring to @RexXiong 's comment, we may need to optimize the RPC perf when there are a huge number of workers (thousands, say) in the future.

Reply:
Given we will have 1000's of workers, we should probably do this as a fast follow @otterc (if not in this PR).

Reply:
Yeah makes sense. I will create a followup

Reply:
@mridulm @waitinfuture I made this change with this PR itself so please take a look.
I realized that for Workers, we are not removing the secret registry from the state. The worker so far didn't maintain any state per application. I think for this we might have to forward the appLost message to workers as well. For this I will create a followup.